diagnostics-extension/src/services/fake_diagnostics.data.ts

```diff
@@ -0,0 +1,281 @@
+  const routineId = generateRandomId();
+  let res: RunRoutineResponse = {id: routineId, status: RoutineStatus.unknown };
+  return res;
+}
+
+export const runLanConnectivityRoutine = async (): Promise<RunRoutineResponse> => {
+  const routineId = generateRandomId();
+  let res: RunRoutineResponse = {id: routineId, status: RoutineStatus.unknown };
+  return res;
+}
+
+export const runMemoryRoutine = async (): Promise<RunRoutineResponse> => {
+  const routineId = generateRandomId();
+  let res: RunRoutineResponse = {id: routineId, status: RoutineStatus.unknown };
+  return res;
+}
+
+export const runNvmeSelfTestRoutine = async (params: DiagnosticsParams) : Promise<RunRoutineResponse> => {
+  const routineId = generateRandomId();
+  let res: RunRoutineResponse = {id: routineId, status: RoutineStatus.unknown };
+  return res;
+}
+
+export const runNvmeWearLevelRoutine = async (params: DiagnosticsParams) : Promise<RunRoutineResponse> => {
+  const routineId = generateRandomId();
+  let res: RunRoutineResponse = {id: routineId, status: RoutineStatus.unknown };
+  return res;
+}
+
+export const runPowerButtonRoutine = async (params: DiagnosticsParams) : Promise<RunRoutineResponse> => {
+  const routineId = generateRandomId();
+  let res: RunRoutineResponse = {id: routineId, status: RoutineStatus.unknown };
+  return res;
+}
+
+export const runSensitiveSensorRoutine = async (): Promise<RunRoutineResponse> => {
+  const routineId = generateRandomId();
+  let res: RunRoutineResponse = {id: routineId, status: RoutineStatus.unknown };
+  return res;
+}
+
+export const runSignalStrengthRoutine = async (): Promise<RunRoutineResponse> => {
+  const routineId = generateRandomId();
+  let res: RunRoutineResponse = {id: routineId, status: RoutineStatus.unknown };
+  return res;
+}
+
+export const runSmartctlCheckRoutine = async (params: DiagnosticsParams) : Promise<RunRoutineResponse> => {
+  const routineId = generateRandomId();
+  let res: RunRoutineResponse = {id: routineId, status: RoutineStatus.unknown };
+  return res;
+}
+
+export const runUfsLifetimeRoutine = async (): Promise<RunRoutineResponse> => {
+  const routineId = generateRandomId();
+  let res: RunRoutineResponse = {id: routineId, status: RoutineStatus.unknown };
+  return res;
+}
+
+export class Routine {
+  private _id: number;
+
+  constructor(id: number) {
+    this._id = id;
+  }
+
+  get id(): number {
+    return this._id;
+  }
+
+  async _genericSendCommand(command: string): Promise<GetRoutineUpdateResponse> {
+    const message = {
+      routineId: this._id,
+      command: command,
+      includeOutput: true,
+    };
+    console.log('Constrcuted diagnostics command', message);
+    const status: GetRoutineUpdateResponse = {
+      progress_percent: 0,
+      output: '',
+      status: RoutineStatus.unknown,
+      status_message: '',
+      user_message: UserMessageType.unknown,
+    };
+    return status;
+  }
+
+  /**
```

Comment:
Is our decision to let the app call `resume`, `stop`, etc? Or is the decision to only give them `manageDiagnosticsRoutine` and let them specify the action themselves.

I think the fake service should be the same as the real implementation.

Reply:
The fake service is the same as the real one now since they both extend class DiagnosticsService, which contains methods getAvailableRoutines(), startRoutine(), stopRoutine(), resumeRoutine(), getRoutineStatus().

And the data flow is that the app should call manageDiagnosticsRoutine() and send the action (e.g. start, stop, resume, etc) as a parameter, and the controller in extension would call diagnosticsService.stopRoutine(), diagnosticsService.resumeRoutine(), etc, to handle the request.